_posts/2020-02-22-01-plus-02-is-not-03.md

```diff
@@ -0,0 +1,330 @@
+
+---
+
+Python is a horrendous joke of a language, you think. Clearly it’s not a language for Real Programmers. It deserves to be resigned to making SEO blogspam generators with that GPT3 nonsense.
+
+So you whip out the trusty Go, merely as a mental exercise of how you would have prevented this disaster.
+
+    package main
+    
+    import (
+            "fmt"
+    )
+    
+    func main() {
+            fmt.Println(0.1 + 0.2 == 0.3)
+    }
+
+`true`. Excellent. And now to finish this up and make the equivalent `nukes_will.go`.
+
+    func main() {
+            var a float64 = 0.1
+            var b float64 = 0.2
+            if a+b == 0.3 {
+                    fmt.Println("I saved the day!")
+            } else {
+                    fmt.Println("Nukes launched!")
+            }
+    }
+
+And then you smugly hit Run.
+
+    Nukes launched!
+
+Horrified, you reach out to the other languages.
+
+    Clojure:
+    (+ 0.1 0.2)   ; 0.30000000000000004
+    
+    Julia:
+    .1 + .2       # 0.30000000000000004
+    
+    Haskell:
+    0.1 + 0.2    -- 0.30000000000000004
+
+What a pity! These were supposed to be sophisticated languages. You found that some languages like PHP and GNU Octave did return `0.3` but, they did not pass the equality test which would have stopped the nukes.
+
+Your stomach grumbles. You haven’t eaten. Well, you’ve found the source, might as well cash in the canned fish you’d get for this contract. You did find the cause, after all.
+
+As you reach for the power button, you feel your fingers tense up. Your mind is not letting you do this.
+
+You *have* to know.[^1]
+
+---
+
+`2020, in a parallel universe with no pandemics or nuclear meltdowns` 
+
+Yogenderprakashan Sarvepalli was hanging out with his buddy Anu at the college library.
+
+“Anu, I have been trying to hack at the source code of *Money Bunny* so that everytime the Bunny hits the Evil Ducks™, I get extra coin.”
+
+Anu rolled her eyes.
+
+“Look,” he sighed, “There’s this handler which changes the game state everytime I punch the Ducks. It should increment my `money` variable by one. It’s doing diddly-squat.”
+
+“Yoge, you know full well I am not going through that Rabbit nonsense.”
+
+“At least have a look at this snippet I made that does the same thing,” he said, tilting his laptop to her.
+
+    float money_inc(money) {
+        return money + 1;
+    }
+    
+    int main() {
+        // say I punched 100,000 evil ducks
+        float money = 0;
+        for (int i = 0; i < 100000; i++) {
+            money = money_inc(money);
+        }
+        printf(money);
+        return 0;
+    }
+
+“Why does this work, but not my progra—”
+
+“Why are you using a float?”
+
+“I don’t know, because that’s what the game does?”
+
+“That’s the problem. I read about this in a blog the other day.”
+
+“Bu-but there’s no problem in *this* code, it prints 100000.”
+
+“Ay you wait da,” she said while ripping a page from her lab observation book.
```

Comment:
Was the "," intentional?

Overall, I really appreciate this post! Nicely done @tfidfwastaken

Reply:
Yep, that's how it's done usually.

Even I was surprised by that, I opened a book lying around, the convention is comma inside quotation.